src/store.rs

```diff
@@ -41,9 +45,47 @@ impl Store {
     pub(crate) fn install_component(&self, cfg: &DownloadCfg) -> Result<Vec<PathBuf>> {
         let component_dir = self.component_dir_path(&cfg.name, &cfg.version);
 
+        // Cache fuels_version for this component if show_fuels_version exists and is true.
+        // We don't want this failure to block installation, so errors are ignored here.
+        if let Ok(c) = Component::from_name(&cfg.name) {
+            if let Some(true) = c.show_fuels_version {
+                if let Err(e) = self.cache_fuels_version(cfg) {
+                    warn!(
+                        "Failed to cache fuels version for component '{}': {}",
+                        cfg.name, e
+                    );
+                };
+            }
+        };
+
         ensure_dir_exists(&component_dir)?;
         download_file_and_unpack(cfg, &component_dir)?;
         // We ensure that component_dir exists above, so its parent must exist here.
         unpack_bins(&component_dir, &component_dir)
     }
+
+    pub(crate) fn cache_fuels_version(&self, cfg: &DownloadCfg) -> Result<()> {
+        let dirname = component_dirname(&cfg.name, &cfg.version);
+
+        if let Ok(fuels_version) = fetch_fuels_version(cfg) {
+            info!("caching fuels version");
+            ensure_dir_exists(&self.path().join(&dirname))?;
+            let fuels_version_path = self.path().join(dirname).join("fuels_version");
```

Comment:
It'd be good to store "fuels_version" as a constant since it's used in multiple places